packages/data/src/registry.js

```diff
@@ -270,15 +280,17 @@ export function createRegistry( storeConfigs = {}, parent = null ) {
 	 * @param {StoreDescriptor} store Store descriptor.
 	 */
 	function register( store ) {
```

Comment:
I possibly mis-understand this! The extra instance from the second bundle wouldn't ever be instantiated. If that's a problem, should we return the already registered store from `registerStoreInstance` in each of these functions? Then, at least the second store instance could get assigned to the one stored in the registry.

I guess that would make the consumer look more like `export const store = register( createReduxStore( ... ) )`. 

Happy to be corrected on this; I don't understand the internals of store registration that well :)

Reply:
The `register` function currently doesn't return anything, and I'm keeping it that way. Would there be anything useful to return? The store descriptor, it's been passed as argument, the caller already has it. Not useful. And the actual instance of the store, it's a private property of the registry, you're supposed to access it solely through the `registry.select` and `registry.dispatch` functions.

The `@automattic/data-stores` package in Calypso uses a pattern with a `register` function, and you can implement it as follows:
```js
function register() {
  const plansStore = createReduxStore( STORE_NAME, ... );
  register( plansStore );
  return plansStore;
}
```
and then use as:
```js
const PLANS_STORE = Plans.register();

registry.select( PLANS_STORE ).getPlanProduct( ... );
```

The legacy `registerStore` function returns a `store` object, but that's all legacy. It's not even the store object returned by `storeDescription.instantiate` and used by the data registry, but the raw Redux store. It's useful for the `use` function for enabling plugins like `persistence`, nothing else.

Reply:
> you're supposed to access it solely through the registry.select and registry.dispatch functions.

Ah, now I understand :) Given this example:

```js
// store
export const fooStore = createReduxStore( ... );
register( fooStore );

// bundle1
import { fooStore as foo1 } from 'foo-store';
useSelect( select => select( foo1 ).selector() );

// bundle2
import { fooStore as foo2 } from 'foo-store';
useSelect( select => select( foo2 ).selector() );
```

If you were to somehow compare `foo1 === foo2`, they would be different objects. I was thinking this could be problematic, but I didn't realize that the important part of the store is just saved in the registry and not in the object returned from `createReduxStore` (and then exported for use.)

But realizing that `export const fooStore` is only relevant in that it lets the registry identify the store based on the store id, I see it won't be an issue.